packages/record/tsconfig.json

```diff
@@ -1,7 +1,8 @@
 {
   "compilerOptions": {
     "baseUrl": ".",
-    "strict": true
+    "strict": true,
+    "strictPropertyInitialization": false
```

Comment:
Why do we need to turn off `strictPropertyInitialization`?

Reply:
This fixes a build error that occurs because of properties that are not necessarily defined in the constructor:
> RollupError: src/recorder.ts:12:11 - error TS2564: Property 's3' has no initializer and is not definitely assigned in the constructor.

One of the errors occurs here in the `Recorder` class:
```javascript
private s3: lookitS3;
```
because `s3` is only initialized if local download is false. This is also fixable in the code - I can instantiate s3 regardless of whether or not it's used, or remove the local download option entirely.

The error also occurs because of these properties in the `TrialRecordExtension` class:
```javascript
private recorder: Recorder;
private filename: string;
```
because the values for these two properties are set in the `on_start` method (which is called at the start of a trial) rather than the extension's constructor (which runs when the experiment first loads). This is less straightforward to fix in the code because I don't think we want to set up the filename or Recorder object until the trial is about to start. But I could try giving them a default value like `null` in the constructor, which would probably remove the need for `strictPropertyInitialization: false`. Do you prefer that solution?

Reply:
I feel like I am being as bit of stickler here, so feel free to push back.  If we build with `strictPropertyInitialization: false` we're hiding the fact that the class variables `s3` and `filename` can be undefined.  So I think it would be best to change the definition of the variables to: 

```typescript
private recorder?: Recorder;
private filename?: string;
```

And not disable `strictPropertyInitialization`.

Reply:
Being a stickler is totally fine, and yes I agree this is a better solution. Thanks!

Reply:
Looking at the code again, filename's type doesn't need to be updated.

Reply:
> Unfortunately, this will mean we'll have to handle were these variables are this variable is undefined and maybe return an error.

Aha, sorry for reading too fast and missing this earlier!

> Looking at the code again, filename's type doesn't need to be updated.

I am still getting the 'not necessarily defined in the constructor' error for `filename` in `packages/record/src/trial.ts`, so I'm going to initialize that variable in the same way (mark it as optional).